cmd/minishift/cmd/openshift/service.go

```diff
@@ -50,25 +54,120 @@ var serviceCmd = &cobra.Command{
 		} else {
-			fmt.Fprintln(os.Stdout, "Opening the service "+namespace+"/"+service+" in the default browser...")
-			browser.OpenURL(url)
+			printToStdOut(serviceSpecs, ip)
 		}
 	},
 }
 
 func init() {
-	serviceCmd.Flags().StringVarP(&namespace, "namespace", "n", "default", "The namespace of the service.")
-	serviceCmd.Flags().BoolVar(&urlMode, "url", false, "Access the service in the command-line console instead of in the default browser.")
+	serviceCmd.Flags().StringVarP(&namespace, "namespace", "n", "", "The namespace of the service.")
+	serviceCmd.Flags().BoolVar(&inbrowser, "in-browser", false, "Access the service in the default browser.")
+	serviceCmd.Flags().BoolVar(&url, "url", false, "Print the service URL to standard output.")
 	serviceCmd.Flags().BoolVar(&https, "https", false, "Access the service with HTTPS instead of HTTP.")
 	OpenShiftCmd.AddCommand(serviceCmd)
 }
+
+func openInBrowser(serviceSpecs []openshift.ServiceSpec, ip string) {
+	serviceURL := getServiceURL(serviceSpecs, ip)
+	fmt.Fprintln(os.Stdout, "Opening the route/NodePort "+serviceURL+" in the default browser...")
+	browser.OpenURL(serviceURL)
+}
+
+func stdOutURL(serviceSpecs []openshift.ServiceSpec, ip string) {
+	serviceURL := getServiceURL(serviceSpecs, ip)
+	fmt.Fprintln(os.Stdout, serviceURL)
+}
+
+func getServiceURL(serviceSpecs []openshift.ServiceSpec, ip string) string {
+	serviceURL := ""
+	namespaceList := isServiceInMultipleNamespace(serviceSpecs, service)
+	if len(namespaceList) == 0 {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Service %s does not exist", service))
+	}
+	if len(namespaceList) > 1 {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Service %s exists in %+v namespaces, you need to chose a specific namespace using -n <namespace_name>", service, namespaceList))
+	}
+
+	for _, serviceSpec := range serviceSpecs {
+		if serviceSpec.Name == service {
+			if serviceSpec.URL != nil {
+				serviceURL = serviceSpec.URL[0]
+				return serviceURL
+
+			} else if serviceSpec.NodePort != "" {
+				nodePortURL := fmt.Sprintf("%s:%s", ip, serviceSpec.NodePort)
+				urlScheme := "http://"
+				if https {
+					urlScheme = "https://"
+				}
+				serviceURL = urlScheme + nodePortURL
+				return serviceURL
+			} else {
+				atexit.ExitWithMessage(1, fmt.Sprintf("Service '%s' in namespace '%s' does not have route associated which can be opened in the browser.", serviceSpec.Name, serviceSpec.Namespace))
+			}
+		}
+	}
+	return serviceURL
+}
+
+func isServiceInMultipleNamespace(serviceSpecs []openshift.ServiceSpec, service string) []string {
+	namespceList := []string{}
+	for _, serviceSpec := range serviceSpecs {
+		if serviceSpec.Name == service {
+			namespceList = append(namespceList, serviceSpec.Namespace)
+		}
+	}
+	return namespceList
+}
+
+func printToStdOut(serviceSpecs []openshift.ServiceSpec, ip string) {
+	var data [][]string
+	var urls, weights string
+
+	for _, serviceSpec := range serviceSpecs {
+		fmt.Println(serviceSpec.Name)
```

Comment:
I think here is an unwanted print statement:

    router
    docker-registry
    kubernetes
    mariadb
    mariadb-ingress
    nodejs-ex
    hawkular-cassandra
    hawkular-cassandra-nodes
    heapster
    hawkular-metrics
    |-----------|-----------|----------|--------------------------------------------------|--------|
    | NAMSEPACE |   NAME    | NODEPORT |                    ROUTE-URL                     | WEIGHT |
    |-----------|-----------|----------|--------------------------------------------------|--------|
    | myproject | nodejs-ex |          | http://nodejs-ex-myproject.192.168.99.145.nip.io |        |
    |-----------|-----------|----------|--------------------------------------------------|--------|